### PR TITLE
noImplicitAny: SingleStat

### PR DESCRIPTION
**Pull request by tskarhed**
#14714
**Special notes for your reviewer**:
A lot of `any` could probably be avoided here, so any feedback on that would be appreciated!

### CODE REVIEW

File: public/app/plugins/panel/singlestat/module.ts [hunk 1]
```diff
@@ -300,7 +303,7 @@ class SingleStatCtrl extends MetricsPanelCtrl {
     this.setValueMapping(data);
   }
 
-  setValueMapping(data) {
+  setValueMapping(data: { value: any; valueFormatted: any; valueRounded: any }) {
```
Comment (dprokop) [inline public/app/plugins/panel/singlestat/module.ts]: This type annotation is repeated. Can you extract that to a separate interface (and keep the definition in this file?). Note that it's better not to name it `ValueMapping` as we already have an interface like that used for value mapping editor  (packages/grafana-ui/src/types/panel.ts)

**Review by dprokop — APPROVED** [review]
Minor thing found, apart from that 👍

File: public/app/plugins/panel/singlestat/module.ts [hunk 2]
```diff
@@ -155,9 +158,9 @@ class SingleStatCtrl extends MetricsPanelCtrl {
     }
 
     tableData.rows.forEach(row => {
-      const datapoint = {};
+      const datapoint: any = {};
 
-      row.forEach((value, columnIndex) => {
+      row.forEach((value: any, columnIndex: any) => {
```
Comment (torkelo) [inline public/app/plugins/panel/singlestat/module.ts]: columnIndex should always be a number

File: public/app/plugins/panel/singlestat/module.ts (hunk 1 above)
Comment (dehrax) [reply public/app/plugins/panel/singlestat/module.ts]: Should I go for the same types on the interface as here, or is `any` sufficient?

https://github.com/grafana/grafana/pull/17616/files/2a869af20da45073555e3773e4c6f5c0bbd9a457#diff-e593190b57ddfe3f482e43e7b8248993R184

Comment (dprokop) [reply public/app/plugins/panel/singlestat/module.ts]: I would opt for specific type, especially that you have created it's basic shape already :)

**Pull request merged**